### ISSUE TITLE
Opening so many processes for library sync cause Intellij to freeze

### ISSUE THREAD

**Issue by memoyil** [KronicDeth/intellij-elixir#1482]
**Describe the bug**
Opening so many processes when you have so many dependencies for library sync cause idea to freeze 

**To Reproduce**
1. Add 90+ dependecy to your project
2. Every project creates its own library sync process
3. Idea cannot cope with them

**Expected behavior**
I would expect serial processing instead of parallelism

**Desktop:**
 - OS: Ubuntu 16.04

**Erlang:**
- Version 21

**Elixir:**
- Version 1.7.4

**Plugin:**
- Version 10.5.0

**Comment by sh41** [comment]
This is the same defect as #3687, which arrived with a screenshot, thread dumps and a second reporter, and was closed when the dependency sync was rewritten.

At 10.5.0 `DepsWatcher` launched a `Task.Backgroundable` titled "Syncing Library in deps/<name>" straight from each VFS callback — no dedup, no debounce, no cap — and each of those tasks then blocked on the EDT to take a write action, doing its VFS traversal and module-root mutation inside the write lock. Ninety deps meant ninety of them, which is what you were watching. (They were IDE background tasks; nothing forked an OS process per dependency.)

Commit f4e399504 deleted `DepsWatcher` and `mix.Watcher` outright and replaced them with a serialized coroutine pipeline: one debounced collector, a set-based pending queue, one drain at a time under a mutex, and a single write action per drain that does model mutation only. A burst of any size now costs one drain. It shipped in 24.0.0.

Closing as a duplicate of #3687 rather than as fixed, so the two reports sit together — that one carries the better evidence and the same resolution.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us why.
